featran-jmh/src/test/scala/com/spotify/featran/jmh/HotEncoderTransformersBenchmark.scala

```diff
@@ -0,0 +1,234 @@
+    val listInput = input.map {
+      case Some(i) => List(i)
+      case _       => List()
+    }
+    val featureSpec = FeatureSpec
+      .of[List[String]]
+      .required(identity)(NHotEncoder("n_hot"))
+    featureSpec.extract(listInput)
+  }
+
+  def oneHotOptional(input: List[String]): FeatureExtractor[List, String] = {
+    val featureSpec = FeatureSpec
+      .of[String]
+      .optional {
+        case "" => None
+        case f  => Option(f)
+      }(OneHotEncoder("one_hot"))
+    featureSpec.extract(input)
+  }
+
+  def oneHotRequired(input: List[String]): FeatureExtractor[List, String] = {
+    val featureSpec = FeatureSpec
+      .of[String]
+      .required(identity)(OneHotEncoder("one_hot"))
+    featureSpec.extract(input)
+  }
+
+  // Lists of Lists for N-Hot
+  private implicit val arbitraryListString: List[List[String]] = List(
+    List(thisString) ++ Seq(""))
+  private implicit val arbitraryListString1000Values: List[List[String]] = List(
+    sample(data, 0.01).toList)
+  private implicit val arbitraryListString1000ValuesWR: List[List[String]] =
+    List(sampleWR(data, 0.01).toList)
+  private implicit val arbitraryListString100000Values: List[List[String]] =
+    List(sample(data, 0.99).toList)
+  private implicit val arbitraryListString100000ValuesWR: List[List[String]] =
+    List(sampleWR(data, 0.99).toList) ++ List(Range(1, 10).map(_ => "").toList)
+
+  // Lists for One-Hot
+  private implicit val arbitraryString: List[String] = List(thisString) ++ Seq(
+    "")
+  private implicit val arbitraryString1000Values: List[String] =
+    sample(data, 0.01).toList
+  private implicit val arbitraryString1000ValuesWR: List[String] =
+    sampleWR(data, 0.01).toList
+  private implicit val arbitraryString100000Values: List[String] =
+    sample(data, 0.99).toList
+  private implicit val arbitraryString100000ValuesWR: List[String] =
+    sampleWR(data, 0.99).toList ++ Seq("")
+
+  // Lists of options for transforming to a lists of lists for N-Hot
+  private implicit val arbitraryListOptional: List[Option[String]] =
+    List(Option(thisString))
+  private implicit val arbitraryListOptional1000Values: List[Option[String]] =
+    sample(data, 0.01).map(i => Option(i)).toList ++ Seq(None)
+  private implicit val arbitraryListOptional1000ValuesWR: List[Option[String]] =
+    sampleWR(data, 0.01).map(i => Option(i)).toList ++ Seq(None)
+  private implicit val arbitraryListOptional100000Values: List[Option[String]] =
+    sample(data, 0.99).map(i => Option(i)).toList ++ Seq(None)
+  private implicit val arbitraryListOptional100000ValuesWR
+    : List[Option[String]] =
+    sample(data, 0.99).map(i => Option(i)).toList ++ Range(1, 10).map(_ => None)
+
+  lazy private val data: Iterator[String] =
+    Range(0, 100000).map(i => s"string$i").toIterator
+  lazy private val thisString = data.toList.head
+
+  val fixedSeed = 235711L
+  def rngSeed: JRandom = new JRandom(fixedSeed)
```

Comment:
I'd call this `rng` not `rngSeed` since it's not really a seed?

Reply:
its reseeding the `rng` in both `def sample` and `def sampleWR`. but it might not be necessary to do this?

Reply:
got it - using the approach about with `vocab` and `s1000`, it makes sense that this is `rng` and not `rngSeed` :)